docs/csharp/how-to/search-strings.md

```diff
@@ -0,0 +1,66 @@
+
+## Does a string contain text?
+
+The <xref:System.String.Contains%2A?displayProperty=nameWithType>, <xref:System.String.StartsWith%2A?displayProperty=nameWithType> and <xref:System.String.EndsWith%2A?displayProperty=nameWithType> methods search a string for specific text. The following example shows each of these methods, and a variation that uses a case insensitive search:
+
+[!code-csharp-interactive[search strings using methods](../../../samples/snippets/csharp/how-to/strings/SearchStrings.cs#1)]
+
+The preceding example demonstrates an important point for using these methods. Searches are **case-sensitive** by default. You use the <xref:System.StringComparison.CurrentCultureIgnoreCase> enum value to specify a case insensitive search.
+
+## Where does the sought text occur in a string?
+
+The <xref:System.String.IndexOf%2A> and <xref:System.String.LastIndexOf%2A> methods also search for text in strings. These methods return the location of the text being sought. If the text isn't found, they return `-1`. The following example shows a search for the first and last occurrence of the word "methods", and displays the text in between.
+  
+[!code-csharp-interactive[search strings for indices](../../../samples/snippets/csharp/how-to/strings/SearchStrings.cs#2)]
+
+## Finding specific text using regular expressions
+
+The <xref:System.Text.RegularExpressions.Regex?displayProperty=nameWithType> class can be used to search strings. These searches can range in complexity from simple to complicated text patterns.
+
+The following code example searches for the word "code" in a sentence, ignoring case. The static method <xref:System.Text.RegularExpressions.Regex.IsMatch%2A?displayProperty=nameWithType> performs the search. You give it the string to search and a search pattern. In this case, a third argument specifies case-insensitive search. For more information, see <xref:System.Text.RegularExpressions.RegexOptions?displayProperty=nameWithType>.  
+  
+[!code-csharp-interactive[Search using regular expressions](../../../samples/snippets/csharp/how-to/strings/SearchStrings.cs#3)]
+  
+> [!TIP]
+> The `string` methods are usually better choices when you are searching for an exact string. The preceding example helps illustrate the syntax for working with regular expressions.
+
+## Does a string follow a pattern?
+
+The following code uses regular expressions to validate the format of each string in an array. The validation requires that each string have the form of a telephone number in which three groups of digits are separated by dashes, the first two groups contain three digits, and the third group contains four digits. The search pattern uses the regular expression `^\\d{3}-\\d{3}-\\d{4}$`. For more information, see [Regular Expression Language - Quick Reference](http://msdn.microsoft.com/library/930653a6-95d2-4697-9d5a-52d11bb6fd4c).
```

Comment:
In the regex section of the .NET standard docs, I've generally tried to add a table that breaks down each regex so that they an be better understood. I'd recommend doing that here. (e.g., ^ -- begin at the beginning of the string, etc.)